include/flatbuffers/stl_emulation.h

```diff
@@ -302,6 +297,146 @@ inline void vector_emplace_back(std::vector<T> *vector, V &&data) {
 
 #endif  // !FLATBUFFERS_CPP98_STL
 
+ // A tag to indicate an empty flatbuffers::optional<T>.
+struct nullopt_t {
+  explicit FLATBUFFERS_CONSTEXPR_CPP11 nullopt_t(int) {}
+};
+
+#ifdef FLATBUFFERS_USE_STD_OPTIONAL
+template<class T>
+using optional = std::optional<T>;
+
+inline constexpr nullopt_t nullopt = std::nullopt;
+
+#else
+// Limited implementation of optional<T>.
```

Comment:
Note here it is intended to be scalar only? Or a `static_assert(std::is_scalar...` ?